### PR TITLE
[Java][jersey2] Use builder pattern for requests

### PR DESCRIPTION
**Pull request by zippolyte**
Add support for group parameters in the jersey2 client.
It follows the pattern used in #1341, with a small adjustment: it only adds path params, and not all required parameters.

This is done in order to generate a more future proof client and prevent potential backward incompatible changes when required query parameters become optional for instance.

The reasoning for adding only path params is that a change (addition/removal of a path param) would be considered a major API contract change, and thus require a major bump of the API version, hence a major of the client as well.

Please let me know what you think.

cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by wing328** [comment]
Looks like it's causing issue as reported by the CI:

```
[ERROR] COMPILATION ERROR : 
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/jersey2/src/test/java/org/openapitools/client/api/FakeApiTest.java:[223,12] method testGroupParameters in class org.openapitools.client.api.FakeApi cannot be applied to given types;
  required: no arguments
  found: java.lang.Integer,java.lang.Boolean,java.lang.Long,java.lang.Integer,java.lang.Boolean,java.lang.Long
  reason: actual and formal argument lists differ in length
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:testCompile (default-testCompile) on project petstore-jersey2: Compilation failure
[ERROR] /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/java/jersey2/src/test/java/org/openapitools/client/api/FakeApiTest.java:[223,12] method testGroupParameters in class org.openapitools.client.api.FakeApi cannot be applied to given types;
[ERROR] required: no arguments
[ERROR] found: java.lang.Integer,java.lang.Boolean,java.lang.Long,java.lang.Integer,java.lang.Boolean,java.lang.Long
[ERROR] reason: actual and formal argument lists differ in length
[ERROR] -> [Help 1]
[ERROR] 
```

Does it work for you locally?

**Comment by zippolyte** [comment]
Friendly ping, I could really use some eyes on this one 🙇

**Comment by zippolyte** [comment]
Bumping again, I would really appreciate some feedback on this 🙇

**Comment by jimschubert** [comment]
I think this looks good, and I personally prefer the builder pattern.

I wonder if this will I impact existing users (consumers of generated code)? I'd so, should we target the 4.3.x branch?

**Comment by zippolyte** [comment]
It should not, because I made the change behind the `x-group-parameters` extension flag. So the code of existing users should not change as long as they haven’t set this in their specifications. 
That said, I don’t have a problem targeting 4.3.x instead if you think it’s safer.

**Comment by zippolyte** [comment]
@jimschubert anything preventing the merge here ?

**Comment by jimschubert** [comment]
No concerns from me as long as there are no breaking changes. I wonder if someone on the technical committee might have a sample app which uses this client and can verify?

cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

**Comment by zippolyte** [comment]
FWIW, if we look at the generated samples, the only places where the code actually changes if for samples explicitly setting `x-group-parameters` to `true`, like here for example: https://github.com/OpenAPITools/openapi-generator/pull/4666/files#diff-8afa920066ea22142ddcda6b8ad0d6ecL768-R940

File: samples/client/petstore/java/jersey2-java6/docs/FakeApi.md [hunk 1]
```diff
@@ -370,8 +382,10 @@ public class Example {
 
         FakeApi apiInstance = new FakeApi(defaultClient);
         FileSchemaTestClass body = new FileSchemaTestClass(); // FileSchemaTestClass | 
-        try {
+        try { 
             apiInstance.testBodyWithFileSchema(body);
+            
+
```
Comment (wing328) [inline samples/client/petstore/java/jersey2-java6/docs/FakeApi.md]: @zippolyte I'll see if I can get rid of these empty lines and the trailing space after `try {` in another PR.

Comment (wing328) [reply samples/client/petstore/java/jersey2-java6/docs/FakeApi.md]: Filed https://github.com/OpenAPITools/openapi-generator/pull/5031

Comment (zippolyte) [reply samples/client/petstore/java/jersey2-java6/docs/FakeApi.md]: Thanks for this ! I always find it tricky to format properly with the templating.

**Pull request merged**